compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -423,6 +423,9 @@ private void setNarrowedTypeInfo(BLangExpression expr, BVarSymbol varSymbol, BTy
         if (expr.getKind() == NodeKind.BINARY_EXPR && ((BLangBinaryExpr) expr).opKind == OperatorKind.NOT_EQUAL) {
             trueType = types.getRemainingType(varSymbol.type, narrowWithType);
             falseType = types.getTypeIntersection(nonLoggingContext, varSymbol.type, narrowWithType, this.env);
+        } else if (expr.getKind() == NodeKind.TYPE_TEST_EXPR && ((BLangTypeTestExpr) expr).isNotIsExpr) {
```

Comment:
```java
trueType = types.getRemainingType(varSymbol.type, narrowWithType);
falseType = types.getTypeIntersection(nonLoggingContext, varSymbol.type, narrowWithType, this.env);
```
L427 to 428 is the same as L424 to 425.
May be we can collapse this `else if`

Reply:
It can be done.  But the condition in if block will be longer. Is it okay?

Reply:
Yes, your way seems cleaner in that case. We can keep it like that.